boot/freeldr/freeldr/arch/i386/xbox/xboxvideo.c

```diff
@@ -101,6 +101,24 @@ XboxVideoClearScreenColor(ULONG Color, BOOLEAN FullScreen)
     }
 }
 
+VOID
+XboxVideoScrollUp(VOID)
+{
+    ULONG BgColor, Dummy;
+    ULONG PixelCount = ScreenWidth * CHAR_HEIGHT *
+                       (((ScreenHeight - 2 * TOP_BOTTOM_LINES) / CHAR_HEIGHT) - 1);
+    PULONG Src = (PULONG)((PUCHAR)FrameBuffer + (CHAR_HEIGHT + TOP_BOTTOM_LINES) * Delta);
+    PULONG Dst = (PULONG)((PUCHAR)FrameBuffer + TOP_BOTTOM_LINES * Delta);
+
+    XboxVideoAttrToColors(ATTR(COLOR_WHITE, COLOR_BLACK), &Dummy, &BgColor);
+
+    for (; PixelCount != 0; PixelCount--)
```

Comment:
```suggestion
    while (PixelCount--)
```
^ I think that's more common pattern for this case.